governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -234,17 +234,48 @@ protected void configure() {
             bindingBuilder.toInstance(mock);
         }
     }
-    
+
     private static class ArchaiusTestConfigOverrideModule extends AbstractModule {
-        private TestCompositeConfig config;
 
-        public ArchaiusTestConfigOverrideModule(TestCompositeConfig config) {
-            this.config = config;
+        private SettableConfig classLevelOverrides;
+        private SettableConfig methodLevelOverrides;
+
+        public ArchaiusTestConfigOverrideModule(SettableConfig classLevelOverrides, SettableConfig methodLevelOverrides) {
+            this.classLevelOverrides = classLevelOverrides;
+            this.methodLevelOverrides = methodLevelOverrides;
+        }
+
+        @Singleton
+        private static class OptionalConfigHolder {
+
+            @com.google.inject.Inject(optional = true)
+            @RuntimeLayer
+            private SettableConfig runtime;
+
+            public SettableConfig get() {
+                if (runtime != null) {
+                    return runtime;
+                } else {
+                    return new DefaultSettableConfig();
+                }
+            }
         }
 
         @Override
         protected void configure() {
-            bind(CompositeConfig.class).annotatedWith(Raw.class).toInstance(config);
+        }
+
+        @Provides
+        @Singleton
+        public TestCompositeConfig compositeConfig(OptionalConfigHolder config) {
+            return new TestCompositeConfig(config.get(), classLevelOverrides, methodLevelOverrides);
+        }
+
+        @Provides
+        @Singleton
+        @Raw
+        public CompositeConfig compositeConfig(TestCompositeConfig config) {
```

Comment:
Why do we need this?  Can't we just continue to use the @Raw CompositeConfig from InternalArchaiusModule?